.github/workflows/publish.yml

```diff
@@ -16,8 +16,24 @@ jobs:
     uses: ./.github/workflows/download_workflow_data.yml
     with:
       run_id: ${{ github.event_name == 'workflow_dispatch' && inputs.run_id || github.event.workflow_run.id }}
+  determine_repository:
```

Comment:
I'm not sure if I like this job. Pushing images to gardenlinux/gardenlinux even if they are nightlies should work as long as we properly tag them as suggested in #2754. In case we need this side step I would love to move it to `publish.yml` as we reference / create repository URLs there anyway.

Reply:
eventually we'll want to go for tagging, but that may have unintended consequences where various tools expect tags, or parts of tags, to be properly formed versions, so introducing e.g. a `~nightly` suffix after the version might cause unexpected side effects. this should be postponed till after the major release.